staging/src/k8s.io/apiserver/pkg/server/filters/cors_test.go

```diff
@@ -25,79 +26,123 @@ import (
-		func() {
-			server := httptest.NewServer(handler)
-			defer server.Close()
-
-			request, err := http.NewRequest("GET", server.URL+"/version", nil)
-			if err != nil {
-				t.Errorf("unexpected error: %v", err)
-			}
-			request.Header.Set("Origin", item.origin)
-			client := http.Client{}
-			response, err = client.Do(request)
-			if err != nil {
-				t.Errorf("unexpected error: %v", err)
-			}
-		}()
-		if item.allowed {
-			if !reflect.DeepEqual(item.origin, response.Header.Get("Access-Control-Allow-Origin")) {
-				t.Errorf("Expected %#v, Got %#v", item.origin, response.Header.Get("Access-Control-Allow-Origin"))
-			}
-
-			if response.Header.Get("Access-Control-Allow-Credentials") == "" {
-				t.Errorf("Expected Access-Control-Allow-Credentials header to be set")
-			}
-
-			if response.Header.Get("Access-Control-Allow-Headers") == "" {
-				t.Errorf("Expected Access-Control-Allow-Headers header to be set")
-			}
-
-			if response.Header.Get("Access-Control-Allow-Methods") == "" {
-				t.Errorf("Expected Access-Control-Allow-Methods header to be set")
-			}
-
-			if response.Header.Get("Access-Control-Expose-Headers") != "Date" {
-				t.Errorf("Expected Date in Access-Control-Expose-Headers header")
-			}
-		} else {
-			if response.Header.Get("Access-Control-Allow-Origin") != "" {
-				t.Errorf("Expected Access-Control-Allow-Origin header to not be set")
-			}
-
-			if response.Header.Get("Access-Control-Allow-Credentials") != "" {
-				t.Errorf("Expected Access-Control-Allow-Credentials header to not be set")
-			}
-
-			if response.Header.Get("Access-Control-Allow-Headers") != "" {
-				t.Errorf("Expected Access-Control-Allow-Headers header to not be set")
-			}
-
-			if response.Header.Get("Access-Control-Allow-Methods") != "" {
-				t.Errorf("Expected Access-Control-Allow-Methods header to not be set")
-			}
-
-			if response.Header.Get("Access-Control-Expose-Headers") == "Date" {
-				t.Errorf("Expected Date in Access-Control-Expose-Headers header")
-			}
+	for _, test := range tests {
+		for _, origin := range test.origins {
+			name := fmt.Sprintf("%s/origin/%s", test.name, origin)
+			t.Run(name, func(t *testing.T) {
+				var handlerInvoked int
+				handler := WithCORS(
+					http.HandlerFunc(func(http.ResponseWriter, *http.Request) {
+						handlerInvoked++
+					}),
+					test.allowedOrigins, nil, nil, nil, "true",
+				)
+				var response *http.Response
+				func() {
+					server := httptest.NewServer(handler)
+					defer server.Close()
+
+					request, err := http.NewRequest("GET", server.URL+"/version", nil)
+					if err != nil {
+						t.Errorf("unexpected error: %v", err)
+					}
+					request.Header.Set("Origin", origin)
+					client := http.Client{}
+					response, err = client.Do(request)
+					if err != nil {
+						t.Errorf("unexpected error: %v", err)
+					}
+				}()
+				if handlerInvoked != 1 {
+					t.Errorf("Expected the handler to be invoked %d time(s), but got: %d", 1, handlerInvoked)
```

Comment:
nit: If we're always comparing `handlerInvoked` to 1, maybe it might be slightly better to have:
```go
t.Errorf("Expected the handler to be invoked once, but got: %d", handlerInvoked)
```

Reply:
yes, done